pkg/issuer/acme/dns/rfc2136/provider.go

```diff
@@ -91,18 +97,22 @@ func (s *Solver) CleanUp(ch *whapi.ChallengeRequest) error {
 }
 
 func (s *Solver) Initialize(kubeClientConfig *restclient.Config, stopCh <-chan struct{}) error {
-	cl, err := kubernetes.NewForConfig(kubeClientConfig)
-	if err != nil {
-		return err
+	// Only start a secrets informerfactory if it is needed (if the solver
+	// is not already initialized with a secrets lister)
+	if s.secretLister == nil {
```

Comment:
A cleaner solution would be to remove the use of Initialize altogether and put everything as opts for `New` I don't want to do it though to limit the amount of change and because I am not entirely sure who might be importing this/[the Solver interface](https://github.com/cert-manager/cert-manager/blob/master/pkg/acme/webhook/webhook.go#L28)/[the webhook implementation of Solver](https://github.com/cert-manager/cert-manager/blob/master/pkg/issuer/acme/dns/webhook/webhook.go#L117)

Reply:
@irbekrm can we expand this comment and say that we want to remove this `s.secretLister == nil` option in the future & that it is deperecated?

Reply:
Thanks @inteon that makes sense 👍🏼 Updated in https://github.com/cert-manager/cert-manager/pull/5691/commits/eaf814cffa91bbe80aaa687934cd00bd54226812